mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/utils/PulsarTopicUtils.java

```diff
@@ -218,7 +218,8 @@ public static String getToConsumerTopicName(String subTopicFilter, String pulsar
         if (subTopicFilter.startsWith(TopicDomain.persistent.value())
                 || subTopicFilter.startsWith(TopicDomain.non_persistent.value())) {
             TopicName topicName = TopicName.get(pulsarTopicName);
-            return TopicName.get(topicName.getDomain().value(), topicName.getTenant(), topicName.getNamespace(),
+            return TopicName.get(topicName.getDomain().value(), topicName.getTenant(),
+                    topicName.getNamespace().replace(topicName.getTenant() + "/", ""),
```

Comment:
It's good to use 'NamespaceName.get(topicName.getNamespace())' instead of replace.

Reply:
TopicName.get(topicName.getDomain().value(), NamespaceName.get(topicName.getNamespace()),
                    URLDecoder.decode(topicName.getLocalName())).toString()

Reply:
`NamespaceName.get("public/default")="public/default"` and we need "default".